pyfr/solvers/base/system.py

```diff
@@ -32,6 +32,7 @@ def __init__(self, backend, rallocs, mesh, initsoln, nregs, cfg):
 
         # Load the elements
         eles, elemap = self._load_eles(rallocs, mesh, initsoln, nregs, nonce)
+        self.eles = eles
```

Comment:
?

Reply:
Making shape and eles instance variables to make them easily available for the pseudointegrator:
   
       for ele in self.system.eles:
            # Allocate storage for previous error
            ele.err_prev = self.backend.matrix(ele._shape, np.ones(ele._shape),
                                               tags={'align'})

            # Append the error kernels to the proxylist
            self.pintgkernels['eles', 'localerrest'].append(errestkern(ele))

Reply:
This has extremely negative consequences on memory use.  The eles are already available, if required, through the elemap member (which you'll note we go to great effort to delete when we're finished with it in order to prevent high memory use).

Reply:
I see. And if eles is not an instance variable it gets deleted automatically when the interpreter exits the constructor, right? I will change this. 

Keeping self._shape is fine?

Reply:
So the objects pointed to by the eles proxylist are kept around for some period of time through ele_map, which is a dictionary which points to the same underlying instances.  However, when we are done with these instances we nuke the ele_map to save memory.

In terms of the shape variable, this is superfluous.  The system class already has an ele_shapes member which is the preferred way of handling this.